modules/tensor_mechanics/src/kernels/OutOfPlanePressure.C

```diff
@@ -0,0 +1,46 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "OutOfPlanePressure.h"
+#include "Function.h"
+
+template<>
+InputParameters validParams<OutOfPlanePressure>()
+{
+  InputParameters params = validParams<Kernel>();
+
+  params.addClassDescription("Apply pressure in the out-of-plane direction in 2D plane stress or generalized plane strain models ");
+  params.addParam<FunctionName>("function","Function used to prescribe pressure");
+  params.addParam<PostprocessorName>("postprocessor","Postprocessor used to prescribe pressure");
+  params.addParam<Real>("factor",1.0,"Scale factor applied to prescribed pressure");
+
+  params.set<bool>("use_displaced_mesh") = true;
+
+  return params;
+}
+
+
+OutOfPlanePressure::OutOfPlanePressure(const InputParameters & parameters)
+  :Kernel(parameters),
+  _postprocessor(parameters.isParamValid("postprocessor") ? &getPostprocessorValue("postprocessor") : NULL),
+  _function(parameters.isParamValid("function") ? &getFunction("function") : NULL),
+  _factor(getParam<Real>("factor"))
+{}
```

Comment:
If you use `params.addParam<FunctionName>("function", "1", "Function used to prescribe pressure");` MOOSE will create a function that returns 1 when not supplied by the user.  Then you can avoid the isParamValid.